src/Solution/ParticleTracker/Particle.f90

```diff
@@ -68,6 +71,7 @@ module ParticleModule
     procedure, public :: get_model_coords
     procedure, public :: load_particle
     procedure, public :: transform => transform_coords
+    procedure, public :: reset_transform
```

Comment:
I pulled out a separate routine to reset the transformation. This seemed cleaner.